src/plugins/intel_gpu/src/graph/impls/ocl/primitive_base.hpp

```diff
@@ -86,7 +87,7 @@ struct typed_primitive_impl_ocl : public typed_primitive_impl<PType> {
 
     template<typename ImplType>
     static std::unique_ptr<primitive_impl> create(const typed_program_node<PType>& arg, const kernel_impl_params& impl_param) {
-        if (impl_param.can_be_optimized()) {
+        if (impl_param.can_be_optimized() && !(impl_param.typed_desc<concatenation>() && impl_param.is_dynamic())) {
```

Comment:
Could you please add note here? 
// concat buffer fusing for dynamic shape is adaptively applied at runtime. So we need to build dynamic impl at build time.

Reply:
Done. Thanks!